core/tests/puppeteer-acceptance-tests/utilities/user/translation-admin.ts

```diff
@@ -72,12 +74,20 @@ export class TranslationAdmin extends BaseUser {
     username: string,
     languageCode: string
   ): Promise<void> {
+    // Wait for the username input field to be visible before typing in a contributor's name.
+    await this.page.waitForSelector(addContributorUsernameInput);
+    // Wait for the contribution rights category dropdown to load before selecting a role.
+    await this.page.waitForSelector(addContributonRightsCategorySelect);
     await this.type(addContributorUsernameInput, username);
     await this.select(
       addContributonRightsCategorySelect,
       translationRightValue
     );
+    // Wait for the language dropdown to appear before selecting the translation language.
+    await this.page.waitForSelector(addContributonRightsLanguageDropdown);
+    // Wait for the submit button to ensure the form is fully loaded before submitting the translation rights.
     await this.select(addContributonRightsLanguageDropdown, languageCode);
+    await this.page.waitForSelector(addContributionRightsSubmitButton);
     await this.clickOn(addContributionRightsSubmitButton);
```

Comment:
can you add a post-check here to ensure the rights have been successfully added?